libs/fluid-elements/button-group/src/lib/button-group/button-group.ts

```diff
@@ -0,0 +1,221 @@
+    </div>`;
+  }
+
+  /** Handles changes in the slot of the template */
+  private _handleSlotChange(): void {
+    this._buttonGroupItems = Array.from(
+      this.querySelectorAll('fluid-button-group-item'),
+    );
+
+    if (this._buttonGroupItems.length > 0) {
+      if (this.checkedId) {
+        const toCheckedItem = this._buttonGroupItems.find(
+          (item) => item.id === this.checkedId,
+        );
+        if (toCheckedItem) {
+          toCheckedItem.checked = true;
+        } else {
+          this.checkedId = null;
+        }
+      } else {
+        this._setFirstTabIndex();
+      }
+    }
+
+    const buttonGroupItem = this._buttonGroupItems.find((item) => item.checked);
+    if (buttonGroupItem) {
+      this.checkedId = buttonGroupItem.id;
+    }
+  }
+
+  /** Called when the checkedChanged event is called. Handles checking/unchecking button-group-items */
+  private _handleCheckedChange(
+    buttonGroupItemEvent: FluidButtonGroupItemCheckedChangeEvent,
+  ): void {
+    for (const buttonGroupItem of this._buttonGroupItems) {
+      if (
+        buttonGroupItem.id !== buttonGroupItemEvent.checkedId &&
+        buttonGroupItem.checked === true
+      ) {
+        buttonGroupItem.checked = false;
+      }
+    }
+    const checkedButtonGroupItem = this._buttonGroupItems.find(
+      (item) => item.checked,
+    );
+    if (checkedButtonGroupItem) {
+      this.checkedId = checkedButtonGroupItem.id;
+    }
+  }
+
+  /** Handler preventing default scrolling */
+  private _handleKeyDown(event: KeyboardEvent): void {
+    if (event.code === SPACE) {
+      event.preventDefault();
+    }
+  }
+
+  /** Handles keyboard support. E.g Arrow navigation and tab support */
+  private _handleKeyUp(event: KeyboardEvent): void {
+    if (event.code === ARROW_RIGHT || event.code === ARROW_LEFT) {
+      let index = this._buttonGroupItems.findIndex(
+        (item) => item.tabIndex === 0,
+      );
+      const oldIndex = index;
+      index = getNextIndex(index, this._buttonGroupItems.length, event.code);
+      this._buttonGroupItems[oldIndex].tabIndex = -1;
+      this._buttonGroupItems[oldIndex].tabbed = false;
+      this._buttonGroupItems[index].tabIndex = 0;
+      this._buttonGroupItems[index].tabbed = true;
+      this._buttonGroupItems[index].shadowRoot?.querySelector('svg')?.focus();
+    }
+  }
+
+  /** Sets the first available button-group-item's tabindex to 0 */
+  private _setFirstTabIndex(): void {
+    for (const item of this._buttonGroupItems) {
+      item.tabIndex = -1;
+    }
+    const toSetTabindexItem = this._buttonGroupItems.find(
+      (item) => !item.disabled,
+    );
+    if (toSetTabindexItem) {
+      toSetTabindexItem.tabIndex = 0;
+    }
+  }
+
+  /** Disables every item in the group */
+  private _setDisabledOnEveryItem(): void {
+    if (this._buttonGroupItems.length > 0) {
+      for (const item of this._buttonGroupItems) {
+        item.disabled = this._disabled;
+      }
+    }
```

Comment:
you don't need the if check because there won't be an iteration if it is an empty array

```suggestion
    for (const item of this._buttonGroupItems) {
      item.disabled = this._disabled;
    }
```